backend/src/main/java/ai/verta/modeldb/advancedService/AdvancedServiceImpl.java

```diff
@@ -553,22 +553,28 @@ public void getHydratedExperimentRunById(
                 .build();
         throw StatusProto.toStatusRuntimeException(status);
       }
-
-      ExperimentRun experimentRun = experimentRunDAO.getExperimentRun(request.getId());
+      String projectId = experimentRunDAO.getProjectIdByExperimentRunId(request.getId());
 
       // Validate if current user has access to the entity or not
       roleService.validateEntityUserWithUserInfo(
-          ModelDBServiceResourceTypes.PROJECT,
-          experimentRun.getProjectId(),
-          ModelDBServiceActions.READ);
+          ModelDBServiceResourceTypes.PROJECT, projectId, ModelDBServiceActions.READ);
+
+      FindExperimentRuns findExperimentRuns =
+          FindExperimentRuns.newBuilder().addExperimentRunIds(request.getId()).build();
+      ExperimentRunPaginationDTO experimentRunPaginationDTO =
+          experimentRunDAO.findExperimentRuns(findExperimentRuns);
+      LOGGER.debug(
+          ModelDBMessages.EXP_RUN_RECORD_COUNT_MSG, experimentRunPaginationDTO.getTotalRecords());
 
       List<HydratedExperimentRun> hydratedExperimentRuns =
-          getHydratedExperimentRuns(Collections.singletonList(experimentRun));
+          getHydratedExperimentRuns(experimentRunPaginationDTO.getExperimentRuns());
 
-      responseObserver.onNext(
-          GetHydratedExperimentRunById.Response.newBuilder()
-              .setHydratedExperimentRun(hydratedExperimentRuns.get(0))
-              .build());
+      GetHydratedExperimentRunById.Response.Builder response =
+          GetHydratedExperimentRunById.Response.newBuilder();
+      if (!hydratedExperimentRuns.isEmpty()) {
```

Comment:
if we are querying by id , seeing multiple experimentRun should at least log a WARN

Reply:
Added WARN log if multiple experiment run found.